File: event-service/src/main/java/org/tggc/eventservice/sender/Sender.java
package org.tggc.eventservice.sender;

import org.tggc.eventservice.model.Event;
import org.tggc.notificationapi.dto.NotificationType;

public interface Sender {

    void send(String email, Event event, NotificationType type);

    NotificationType getNotificationType();
}

File: event-service/src/main/kotlin/org/tggc/eventservice/sender/Sender.kt
package org.tggc.eventservice.sender

import org.tggc.eventservice.model.Event
import org.tggc.notificationapi.dto.NotificationType
import reactor.core.publisher.Mono

interface Sender {

    fun send(email: String, event: Event, type: NotificationType): Mono<Void>

    val notificationType: NotificationType
}
